### PR TITLE
Bugfix/860-861 Fix Unhandled Exceptions

### PR DESCRIPTION
**Pull request by BohdanZaliubovskyi**
Related issues: #860, #861 

fixed catch of the generated exception

### CODE REVIEW

File: CharlieBackend.Panel/Utils/ApiUtil.cs [hunk 1]
```diff
@@ -18,6 +21,13 @@ public ApiUtil(IHttpUtil httpUtil)
             _httpUtil = httpUtil;
         }
 
+        private async Task EnsureSuccessStatusCode(HttpResponseMessage httpResponse)
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/Utils/ApiUtil.cs]: In yesterday's meeting looks like we agreed to add the async suffix to async methods, didn`t it?

File: CharlieBackend.Panel/Utils/ApiUtil.cs [hunk 2]
```diff
@@ -18,6 +21,13 @@ public ApiUtil(IHttpUtil httpUtil)
...
+        {
+            string message = await _httpUtil.EnsureSuccessStatusCode(httpResponse);
+            if(message != string.Empty)
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/Utils/ApiUtil.cs]: Maybe IsNullOrEmpty would be better here? I know that this method returns string.empty, but still

File: CharlieBackend.Panel/Utils/HttpUtil.cs [hunk 3]
```diff
@@ -76,16 +76,22 @@ public async Task<HttpResponseMessage> DeleteAsync(string url)
...
-        public async Task EnsureSuccessStatusCode(HttpResponseMessage httpResponse)
+        public async Task<string> EnsureSuccessStatusCode(HttpResponseMessage httpResponse)
         {
             if (!httpResponse.IsSuccessStatusCode)
             {
                 string apiStringResponse = await httpResponse.Content.ReadAsStringAsync();
 
-                var apiResponseMessage = JsonConvert.DeserializeObject<ErrorDto>(apiStringResponse).Error.Message;
+                string apiResponseMessage;
+                if (apiStringResponse != "")
```
Comment (PavelTsyposhchuk) [inline CharlieBackend.Panel/Utils/HttpUtil.cs]: IsNullOrEmptyOrWhiteSpace definetly would be better

**Review by PavelTsyposhchuk — CHANGES_REQUESTED** [review]

File: CharlieBackend.Panel/Utils/HttpUtil.cs [hunk 4]
```diff
@@ -76,16 +76,22 @@ public async Task<HttpResponseMessage> DeleteAsync(string url)
...
+                    apiResponseMessage = JsonConvert.DeserializeObject<ErrorDto>(apiStringResponse).Error.Message;
+                else
+                    apiResponseMessage = ErrorsConstants.EmptyApiStringResponse;
 
-                throw new HttpStatusException(httpResponse.StatusCode, apiResponseMessage);
+                return apiResponseMessage;
```
Comment (filinskyi) [inline CharlieBackend.Panel/Utils/HttpUtil.cs]: curly brackets for "if else" body, 
if condition replace "" to string.Empty,
and it is possible to simplify it as inline return or assigned to the variable "apiResponseMessage" 
```
return apiStringResponse != string.Empty 
          ? JsonConvert.DeserializeObject<ErrorDto>(apiStringResponse).Error.Message
          : ErrorsConstants.EmptyApiStringResponse;
```

**Review by filinskyi — APPROVED** [review]
LGTM

**Review by PavelTsyposhchuk — APPROVED** [review]

**Review by DevPew01 — APPROVED** [review]

**Review by andy-dendy-repo — APPROVED** [review]

**Review by Ssashaa — APPROVED** [review]

**Pull request merged**